client/src/components/Facts.js

```diff
@@ -0,0 +1,112 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import moment from "moment"
+
+function Facts(props) {
+  //declares state to hold entries once retrieved from database
+  const [entries, setEntries] = useState();
+
+  //access query from url, removing ? from front of query
+  let query = window.location.search.slice(1);
+
+  //sanitize query and assign appropriate values to category and user-input
+  let queryArray = query.split("=");
+  let category = queryArray[0];
+  let input = queryArray[1];
+
+  //pulls entries from database upon load
+  useEffect(() => {
+    //guard clause to check if entries is already populated
+    if (!entries) {
+      //if there are query params in the url the results should be filtered, fetch filtered posts
+      if (category && input) {
+        fetch(`/filterPosts/${category}/${input}`)
+          .then((res) => res.json())
+          //sets resulting array in state as entries
+          .then((entryInfo) => {
+            let resultsArray = [];
+            entryInfo.forEach((post) => {
+              resultsArray.push(post);
+            });
+            setEntries(resultsArray);
+          });
+      }
+      //if there are no query params in the url there is no filtering, fetch all posts
+      else {
+        fetch("/allposts")
+          .then((res) => res.json())
+          .then((entryInfo) => {
+            let resultsArray = [];
+            entryInfo.forEach((post) => {
+              resultsArray.push(post);
+            });
+            setEntries(resultsArray);
+          });
+      }
+    }
+  });
+
+  //initiates array of categories to be used in category display
+  let arrayOfCategories = [
+    "javascript",
+    "json",
+    "html",
+    "css",
+    "frontend",
+    "backend",
+    "fullstack",
+    "databases",
+  ];
+
+  // each entry is set up as a link to entry page with title, date, and categories
+  return (
+    <div id="facts-page-container">
+      {/* filter with user input options */}
+      <div id="search-container">
+        <form id="search-form" method="POST" action="search">
+          <select id="filter-options" name="category">
+            <option value="pick-a-filter">Filter by</option>
+            <option value="title">Title</option>
+            <option value="content">Content</option>
+            <option value="categories">Category</option>
+          </select>
+          <input id="user-input" name="input" type="text" />
+          <input id="filter-button" type="submit" value="Filter" />
+        </form>
+
+        {/* if results are filtered button shows offering option to remove filters */}
+        {query && <Link to={"/facts"}><button onClick ="window.location.reload()"  id="remove-all-filters">Remove All Filters</button></Link>}
+      </div>
+
+      {/* displays list of all posts and attributes */}
+      <div id="list-of-entries">
+        {entries &&
+          entries.map((entry, index) => {
```

Comment:
The only "complaint" I have for you is that the console is displaying an error `Array.prototype.map() expects a value to be returned at the end of arrow function  array-callback-return`.
This is happening because data.map function (check Array.prototype.map specification) converts one array to a new array. The conversion (mapping) is defined by the argument of data.map, also called the callback function. In your case, the callback function is the arrow function  =>. The callback function of data.map must return a value. By returning a value for each item of the array is how data.map defines the mapping.

But in your case the callback function does not return anything. Your intention is not to do any kind of mapping, but just to create jsx to display on the frontend. So in your case you can use data.forEach as you don't use data.map functionality.

Basically, changing `.map` on lines 84 and 98 to `.forEach` will both keep your function working exactly like it is now, as well as make React happy.